File: LeetCode-Problems/Easy/Decompress-RLE-List.py
# Python
class Solution(object):
    def decompressRLElist(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        returnedList = list()
        pair = [0, 1]
        x = 0
        while (x < len(nums)):
            pair[0] = nums[x]
            pair[1] = nums[x+1]
            x = x+2
            for y in range (0, pair[0]):
                returnedList.append(pair[1])
        return returnedList

File: LeetCode-Problems/Easy/Decompress-RLE-List.java
//Java
class Solution {
    public int[] decompressRLElist(int[] nums) {
        int[] currentPair = new int[2];
        ArrayList<Integer> returnedList = new ArrayList<Integer>();
        for (int i = 0; i < nums.length; i+=2){
            currentPair[0] = nums[i];
            currentPair[1] = nums[i+1];
            for (int j = 0; j < currentPair[0]; j++){
                returnedList.add(currentPair[1]); 
            }
        }
        int[] returnedIntList = new int[returnedList.size()];
        for (int i = 0; i < returnedList.size(); i++){
            returnedIntList[i] = returnedList.get(i);
        }
        return returnedIntList;
    }
}
